posts/2023-11-05-telegram-bot-for-file-sharing-using-quarkus.adoc

```diff
@@ -0,0 +1,495 @@
+```bash
+quarkus create app com.example:file-sharing-bot \
+--extension=quarkus-mongodb-panache,quarkus-resteasy,quarkus-quinoa \
+--gradle
+```
+
+We can already run our application with the default implementation. To do so, navigate to the root directory and execute the command to start our application:
+```bash
+quarkus dev
+```
+
+Now, you can open the recommended welcome page at http://localhost:8080/
+
+image::welcome_page.png[,600]
+
+and http://localhost:8080/hello default controller method
+
+image::controller_hello.png[,600]
+
+and http://localhost:8080/quinoa.html default frontend implementation
+
+image::quinoa_default.png[,600]
+
+It was quite easy, right?
+
+Before we dive into coding the bot's logic, we need to register our bot and obtain a username and token. To do this, please follow the official instructions provided here: https://core.telegram.org/bots/tutorial.
+
+It's worth noting that there isn't even a main class like we're used to seeing in the source code, and which is sometimes considered boilerplate. However, let's go ahead and create one explicitly now:
+[source,java]
+----
+package com.example;
+
+import io.quarkus.runtime.Quarkus;
+import io.quarkus.runtime.QuarkusApplication;
+import io.quarkus.runtime.annotations.QuarkusMain;
+import org.jboss.logging.Logger;
+import org.telegram.telegrambots.bots.TelegramLongPollingBot;
+import org.telegram.telegrambots.meta.TelegramBotsApi;
+import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
+import org.telegram.telegrambots.updatesreceivers.DefaultBotSession;
+
+@QuarkusMain
+public class Main {
+
+    public static void main(String... args) {
+        Quarkus.run(MyApp.class, args);
+    }
+
+    public static class MyApp implements QuarkusApplication {
+        private static final Logger LOGGER = Logger.getLogger(MyApp.class);
+        private final TelegramLongPollingBot telegramLongPollingBot;
+
+        public MyApp(TelegramLongPollingBot telegramLongPollingBot) {
+            this.telegramLongPollingBot = telegramLongPollingBot;
+        }
+
+        @Override
+        public int run(String... args) {
+            try {
+                TelegramBotsApi telegramBotsApi = new TelegramBotsApi(DefaultBotSession.class);
+                telegramBotsApi.registerBot(telegramLongPollingBot);
+            } catch (TelegramApiException e) {
+                LOGGER.errorf(e, e.getMessage());
+            }
+            Quarkus.waitForExit();
+            return 0;
+        }
+    }
+}
+----
+
+Dependency injection in Quarkus is a CDI-based dependency injection - https://quarkus.io/guides/cdi. In the case of only one defined constructor, Quarkus uses constructor injection. To satisfy the MyApp constructor parameter TelegramLongPollingBot, let's create the corresponding bean using the [.yellow]#@Produce# annotation.
+
+To satisfy the constructor parameter  of  let's create a corresponding bean using the method return type by [.yellow]#@Produce# annotation:
```

Comment:
```suggestion
To satisfy the constructor parameter, let's create a corresponding bean using the [.yellow]#@Produce# annotation:
```
or maybe you forgot a word, after `of`?
Another thing, I find it confusing that you say that the annotation is the method return type. They are different things